### PR TITLE
feat(tree): [tree] virtual scroll, collapse and filter functions

### PR DESCRIPTION
**Pull request by necoxrr**
# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-vue/blob/dev/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

### CODE REVIEW

File: packages/renderless/src/tree/index.ts [hunk 1]
```diff
@@ -440,15 +442,21 @@ export const filter =
     if (!props.filterNodeMethod) {
       throw new Error('[Tree] filterNodeMethod is required when filter')
     }
-
+    console.log('值', api);
```
Comment (ErKeLost) [inline packages/renderless/src/tree/index.ts:445]: 记得删除 log

File: packages/renderless/src/virtual-scroll/vue.ts [hunk 2]
```diff
@@ -61,23 +61,25 @@ export const renderless = (props, { reactive, nextTick, watch, onMounted, ref },
     state.temporary.prerender = true
     api.handleScroll()
   }
-
-  // 组件挂载后加载滚动事件
-  onMounted(() => {
+  const handle = () => {
```
Comment (ErKeLost) [inline packages/renderless/src/virtual-scroll/vue.ts:64]: handle 这个函数名 应该换成更语义化的方法名

**Review by ErKeLost — APPROVED** [review]

File: packages/renderless/src/tree/index.ts (hunk 1 above)
Comment (necoxrr) [reply packages/renderless/src/tree/index.ts:445]: 优化PR上改掉

File: packages/renderless/src/virtual-scroll/vue.ts (hunk 2 above)
Comment (necoxrr) [reply packages/renderless/src/virtual-scroll/vue.ts:64]: 优化PR上更改

**Review by kagol — APPROVED** [review]

**Comment by kagol** [comment]
@necoxrr 麻烦解决下冲突哈~

**Pull request closed without merging**